sql/src/main/java/io/crate/jobs/NestedLoopContext.java

```diff
@@ -0,0 +1,224 @@
+        }
+    }
+
+    @Override
+    public void close() {
+        if (!closed.getAndSet(true)) {
+            if (activeSubContexts.get() == 0) {
+                callContextCallback();
+            } else {
+                if (leftDownstreamContext != null) {
+                    leftDownstreamContext.close();
+                }
+                if (rightDownstreamContext != null) {
+                    rightDownstreamContext.close();
+                }
+            }
+        }
+    }
+
+    @Override
+    public void kill() {
+        if (!killed.getAndSet(true)) {
+            if (activeSubContexts.get() == 0) {
+                callContextCallback();
+            } else {
+                if (leftDownstreamContext != null) {
+                    leftDownstreamContext.kill();
+                }
+                if (rightDownstreamContext != null) {
+                    rightDownstreamContext.kill();
+                }
+            }
+        }
+    }
+
+    @Override
+    public String name() {
+        return nestedLoopNode.name();
+    }
+
+    @Override
+    public boolean isKilled() {
+        return killed.get();
+    }
+
+    @Override
+    public PageDownstreamContext pageDownstreamContext(int inputId) {
+        assert inputId < 2 : "Only 0 and 1 inputId's supported";
+        if (inputId == 0) {
+            return leftDownstreamContext;
+        }
+        return rightDownstreamContext;
+    }
+
+    private void callContextCallback() {
+        if (contextCallbacks.isEmpty()) {
+            return;
+        }
+        if (activeSubContexts.get() == 0) {
+            for (ContextCallback contextCallback : contextCallbacks) {
+                contextCallback.onClose(null, ramAccountingContext.totalBytes());
+            }
+            ramAccountingContext.close();
+        }
+    }
+
+    public void interruptIfKilled() {
+        if (killed.get()) {
+            throw new CancellationException();
+        }
+    }
+
+    private PageDownstreamContext createPageDownstreamContext(MergeNode node) {
+        Tuple<PageDownstream, FlatProjectorChain> pageDownstreamProjectorChain =
+                pageDownstreamFactory.createMergeNodePageDownstream(
+                        node,
+                        nestedLoopOperation,
+                        ramAccountingContext,
+                        Optional.of(threadPool.executor(ThreadPool.Names.SEARCH)));
+        StreamerVisitor.Context streamerContext = streamerVisitor.processPlanNode(node);
+        PageDownstreamContext pageDownstreamContext = new PageDownstreamContext(
+                node.name(),
+                pageDownstreamProjectorChain.v1(),
+                streamerContext.inputStreamers(),
+                ramAccountingContext,
+                node.numUpstreams());
+
+        FlatProjectorChain flatProjectorChain = pageDownstreamProjectorChain.v2();
+        if (flatProjectorChain != null) {
+            flatProjectorChain.startProjections(pageDownstreamContext);
```

Comment:
this starts projections, which should be done on context.start() not upon instantiation of the context.